internal/platform/open_file_windows.go

```diff
@@ -140,6 +140,10 @@ func open(path string, mode int, perm uint32) (fd syscall.Handle, err error) {
 			}
 		}
 	}
+	if createmode == syscall.OPEN_EXISTING && access == syscall.GENERIC_READ {
```

Comment:
lifted from the change made in Go 1.20 https://github.com/golang/go/blob/go1.20/src/syscall/syscall_windows.go#L308-L379 and I guess that's why we haven't encountered it before.